### PR TITLE
Add X_API_KEY environment variable and with_api_key middleware

### PR DESCRIPTION
**Pull request by mukeshkuiry**
This pull request adds the X_API_KEY environment variable and the with_api_key middleware to the codebase. These changes allow for secure authentication and authorization when making API requests. #45 Issue

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -35,7 +36,8 @@ async fn main() -> Result<(), Box<dyn Error>> {
         .merge(routes::user_routes::routes(State(app_state.clone())))
         .merge(routes::password_routes::routes(State(app_state.clone())))
         .merge(routes::session_routes::routes(State(app_state.clone())))
-        .layer(middleware::map_response(main_response_mapper));
+        .layer(middleware::map_response(main_response_mapper))
+        .route_layer(middleware::from_fn(with_api_key));
```
Comment (Rajdip019) [inline src/main.rs:40]: What's the difference between`layer` and `router_layer`

Comment (mukeshkuiry) [reply src/main.rs]: In Axum, `layer` applies middleware to the entire router, affecting all routes within that router, while `route_layer` applies middleware to a specific route, affecting only that route. Use `layer` for global middleware and `route_layer` for route-specific middleware.

But we can use layer also here for consistancy, so i changed it to layer only.

File: docs/local-setup/readme.md [hunk 2]
```diff
@@ -1,30 +1,35 @@
...
 
-
 SMTP servers require authentication to ensure that only authorized users can send emails. For generating `EMAIL_PASSWORD`, Visit this [link](https://support.google.com/mail/thread/205453566/how-to-generate-an-app-password?hl=en).
 `SMTP_DOMAIN = smtp.gmail.com` as we are using GMAIL as a Mail Provider.
 
+You have to add a random Access Key `X_API_KEY` for secure authentication of the endpoints.
+Alternatively you can use this command to generate a random `API KEY`
+
+```shell
+gpg --gen-random --armor 1 18
```
Comment (Rajdip019) [inline docs/local-setup/readme.md:28]: Use open SSL to generate a secret. 

maybe `openssl rand -base64 32` as we are using open SSL for all of these. @mukeshkuiry

Comment (mukeshkuiry) [reply docs/local-setup/readme.md]: Changed

**Review by Rajdip019 — APPROVED** [review]

**Pull request merged**